File: cpp/Minimum_Deletions_to_Make_String_Balanced_1653.cpp
class Solution {
public:

    int minimumDeletions(string s) {
        int nextA = 0, nextB = 0;
        for (int i = s.size() - 1; i >= 0; --i) {
            if (s[i] == 'a') {
                nextB += 1;
            }
            else {
                nextA = min(nextA + 1, nextB);
            }
        }
        return min(nextA, nextB);
    }
};

File: python/Minimum_Deletions_to_Make_String_Balanced_1653.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        nextA, nextB = 0, 0
        for i in range(len(s) - 1, -1, -1):
            if s[i] == 'a':
                nextB += 1
            else:
                nextA = min(nextA + 1, nextB)
        return min(nextA, nextB)
